app/auth/Register/page.jsx

```diff
@@ -1,130 +1,161 @@
-            </Container>
-    );
-};
+  const handleShowPassword = () => setShowPassword((show) => !show)
+  const handleShowConfirmPassword = () =>
+    setShowConfirmPassword((show) => !show)
 
-export default RegisterForm;
+  const handleSubmit = async (e) => {
+    e.preventDefault()
+    if (!isLogin) {
+      try {
+        await register(formData)
+        router.push('/auth/login')
+      } catch (error) {
+        console.log(error.message)
+      }
+    }
+  }
+
+  return (
+    <Container maxWidth="sm">
+      <Typography variant="h4" sx={Style.Title}>
+        Welcome to CoursePRN
+      </Typography>
+      <Typography variant="subtitle1" sx={Style.SubTitle}>
+        Sign up to receive offers and manage your courses better
+      </Typography>
+      <form onSubmit={handleSubmit}>
+        <TextField
+          label="Username"
+          variant="outlined"
+          name="username"
+          fullWidth
+          margin="normal"
+          onChange={handleInputChange}
+          value={formData.username}
+        />
+        <TextField
+          label="Full Name"
+          variant="outlined"
+          name="fullName"
+          fullWidth
+          margin="normal"
+          onChange={handleInputChange}
+          value={formData.fullName}
+        />
+        <TextField
+          label="Email"
+          variant="outlined"
+          name="email"
+          fullWidth
+          margin="normal"
+          onChange={handleInputChange}
+          value={formData.email}
+        />
+        <FormControl fullWidth variant="outlined" margin="normal">
+          <InputLabel htmlFor="password">Password</InputLabel>
+          <OutlinedInput
+            id="password"
+            type={showPassword ? 'text' : 'password'}
+            name="password"
+            onChange={handleInputChange}
+            value={formData.password}
+            endAdornment={
+              <InputAdornment position="end">
+                <IconButton onClick={handleShowPassword} edge="end">
+                  {showPassword ? <Visibility /> : <VisibilityOff />}
+                </IconButton>
+              </InputAdornment>
+            }
+            label="Password"
+          />
+        </FormControl>
+        <FormControl fullWidth variant="outlined" margin="normal">
+          <InputLabel htmlFor="confirmPassword">Confirm Password</InputLabel>
+          <OutlinedInput
+            id="confirmPassword"
+            type={showConfirmPassword ? 'text' : 'password'}
+            name="confirmPassword"
+            onChange={handleInputChange}
+            value={formData.confirmPassword}
+            endAdornment={
+              <InputAdornment position="end">
+                <IconButton onClick={handleShowConfirmPassword} edge="end">
+                  {showConfirmPassword ? <Visibility /> : <VisibilityOff />}
+                </IconButton>
+              </InputAdornment>
+            }
+            label="Confirm Password"
+          />
```

Comment:
I see that the password and confirmPassword fields are the same so I think you can create them as a common component to reduce code